src/quadratic_newton.jl

```diff
@@ -0,0 +1,217 @@
+
+"""
+    Semi-smooth Newton method (Algorithm 2 of Lorenz et al. 2019) for solving quadratically regularised optimal transport
+
+    See also: [`QuadraticOTNewton`](@ref), [`quadreg`](@ref)
+
+    # References
+    
+    Lorenz, Dirk A., Paul Manns, and Christian Meyer. [Quadratically regularized optimal transport.](https://doi.org/10.1007/s00245-019-09614-w) Applied Mathematics & Optimization 83.3 (2021): 1919-1949.
+"""
+struct QuadraticOTNewton{T<:Real,K<:Real,D<:Real} <: QuadraticOT
+    θ::T
+    κ::K
+    δ::D
+end
+
+"""
+    QuadraticOTNewton(θ = 0.1, κ = 0.5, δ = 1e-5)
+
+    Semi-smooth Newton method (Algorithm 2 of Lorenz et al. 2019) with Armijo parameters `θ` and `κ`, and conjugate gradient regularisation `δ`. 
+    
+    See also: [`QuadraticOTNewton`](@ref), [`quadreg`](@ref)
+"""
+function QuadraticOTNewton(θ=0.1, κ=0.5, δ=1e-5)
```

Comment:
Did you want to write
```suggestion
function QuadraticOTNewton(; θ=0.1, κ=0.5, δ=1e-5)
```
? Currently this leads to a stack overflow if all values are specified.

Reply:
My bad, yes you're right!